bot/exts/evergreen/fun.py

```diff
@@ -87,6 +88,91 @@ def conversion_func(text: str) -> str:
             converted_text = f">>> {converted_text.lstrip('> ')}"
         await ctx.send(content=converted_text, embed=embed)
 
+    @commands.group(name="caesarcipher", aliases=("caesar",))
+    async def caesarcipher_group(self, ctx: Context) -> None:
+        """
+        Translates a message using the Caesar Cipher.
+
+        See `decrpyt`, `encrypt`, and `info` subcommands.
+        """
+        if ctx.invoked_subcommand is None:
+            await self.bot.get_cog("Help").new_help(ctx, "caesarcipher")
+
+    @caesarcipher_group.command(name="info")
+    async def caesarcipher_info(self, ctx: Context) -> None:
+        """Information about the Caesar Cipher."""
+        with open("bot\\resources\\evergreen\\caesar_info.json", "r") as f:
+            data = json.load(f)
+
+        embed = Embed(
+            title=data["title"],
+            description="".join(data["description"]),
+            colour=Colours.dark_green,
+        )
+
+        await ctx.send(embed=embed)
+
+    @staticmethod
+    async def _caesar_cipher(ctx: Context, offset: int, text: str) -> None:
+        """
+        Given an integer `offset`, translates and sends the given `text`.
+
+        A positive `offset` will cause the letters to shift right, while
+        a negative `offset` will cause the letters to shift left.
+
+        Also accepts a valid Discord Message ID or link.
+        """
+        def caesar_func(text: str) -> str:
+            """Implements a lazy Caesar Cipher algorithm."""
+            for char in text:
+                if not char.isascii() or not char.isalpha() or char.isspace():
+                    yield char
+                    continue
+                case_start = 65 if char.isupper() else 97
+                yield chr((ord(char) - case_start + offset) % 26 + case_start)
+
+        def conversion_func(text: str) -> str:
+            """Encrypts the given string using the Caesar Cipher."""
+            return "".join(caesar_func(text))
+
+        text, embed = await Fun._get_text_and_embed(ctx, text)
+        # Convert embed if it exists
+        if embed is not None:
+            embed = Fun._convert_embed(conversion_func, embed)
+        converted_text = conversion_func(text)
+        # Don't put >>> if only embed present
+        if converted_text:
+            converted_text = f">>> {converted_text.lstrip('> ')}"
+        await ctx.send(content=converted_text, embed=embed)
+
+    @caesarcipher_group.command(name="encrypt", aliases=("rightshift", "rshift"))
+    async def caesarcipher_encrypt(self, ctx: Context, offset: int, *, text: str) -> None:
```

Comment:
I think better is use `Union[str, Message]` converter instead converting this manually. Same in L163.